### PR TITLE
new Sensors Applet

### PR DESCRIPTION
**Pull request by memoryleakx**
Hi,I have created a new (CPU Temperature ) Sensors Applet.

### CODE REVIEW

File: panel/applets/sensors/SensorsApplet.vala [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                                                                    out ls_status);
+                  }
+                  else
+                  {
+                          m_exec = Process.spawn_command_line_sync ("sensors -f",
+                                                                    out ls_stdout,
+                                                                    out ls_stderr,
+                                                                    out ls_status);
+                  }
+                  if (ls_stdout != null)
```
Comment (ikeydoherty) [inline panel/applets/sensors/SensorsApplet.vala]: This level of indentation makes me think you actually have a bug here, it shouldn't be this complex. Fail first, fail quickly.

File: panel/applets/sensors/SensorsApplet.vala [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+          string ls_stdout;
+          string ls_stderr;
+          int ls_status;
+          string editStr = "";
+
+          string [] lines;
+          string [] elines;
+          string [] e2lines;
+
+          lock (m_lock)
```
Comment (ikeydoherty) [inline panel/applets/sensors/SensorsApplet.vala]: Lock and `spawn_command_line_sync` ? You're blocking the whole panel thread and breaking the desktop here. Use concurrency if necessary

File: panel/applets/sensors/SensorsApplet.vala [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+          m_widget = new Gtk.EventBox();
+          m_label = new Gtk.Label(m_resultString);
+          m_widget.add(m_label);
+          m_label.show();
+
+          create_menuMain();
+          m_popover = new Gtk.Popover.from_model(m_widget, m_menuMain);
+          m_popover.get_child().show_all();
+
+          m_widget.button_press_event.connect((e)=> {
```
Comment (ikeydoherty) [inline panel/applets/sensors/SensorsApplet.vala]: Inconsistent styling.

 * 4 spaces
 * No space between words and `()`
 * Inconsistent function naming, switches between camel case and lowercase  splitting

File: panel/applets/sensors/SensorsApplet.vala [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+                          m_isCelsius = false;
+                          break;
+                  }
+          }
+  }
+
+  public void create_menuMain()
+  {
+          m_menuMain = new GLib.Menu();
+          m_menuMain.append ("Celsius", "celsius");
```
Comment (ikeydoherty) [inline panel/applets/sensors/SensorsApplet.vala]: Spaces before `(`, need to go.

File: panel/applets/sensors/SensorsApplet.vala [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+          var celsius_action = new SimpleAction ("celsius", null);
+          celsius_action.activate.connect (this.celsius_cb);
+
+          var fahrenheit_action = new SimpleAction ("fahrenheit", null);
+          fahrenheit_action.activate.connect (this.fahrenheit_cb);
+  }
+
+
+  void celsius_cb (SimpleAction simple, Variant ? parameter) {
+          m_isCelsius = true;
```
Comment (ikeydoherty) [inline panel/applets/sensors/SensorsApplet.vala]: Hungarian notation..?

File: panel/applets/sensors/SensorsApplet.vala [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+
+          var celsius_action = new SimpleAction ("celsius", null);
+          celsius_action.activate.connect (this.celsius_cb);
+
+          var fahrenheit_action = new SimpleAction ("fahrenheit", null);
+          fahrenheit_action.activate.connect (this.fahrenheit_cb);
+  }
+
+
+  void celsius_cb (SimpleAction simple, Variant ? parameter) {
```
Comment (ikeydoherty) [inline panel/applets/sensors/SensorsApplet.vala]: No space after nullable param

File: panel/applets/sensors/SensorsApplet.vala [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+  }
+
+  private bool getSensors()
+  {
+          string ls_stdout;
+          string ls_stderr;
+          int ls_status;
+          string editStr = "";
+
+          string [] lines;
```
Comment (ikeydoherty) [inline panel/applets/sensors/SensorsApplet.vala]: Should be `string[]`, no spaces.

File: panel/applets/sensors/SensorsApplet.vala [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+                                                                    out ls_stdout,
+                                                                    out ls_stderr,
+                                                                    out ls_status);
+                  }
+                  if (ls_stdout != null)
+                  {
+                          lines = ls_stdout.split("\n");
+                          for (int i =0; i < lines.length; i++)
+                          {
+                                  if ("Core" in lines[i])
```
Comment (ikeydoherty) [inline panel/applets/sensors/SensorsApplet.vala]: JS Style equality comparison, you're comparing a current _thing_ to a constant, so the constant should be the Y param

File: panel/applets/sensors/SensorsApplet.vala [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+
+          string [] lines;
+          string [] elines;
+          string [] e2lines;
+
+          lock (m_lock)
+          {
+                  if (m_isCelsius == true)
+                  {
+                          m_exec = Process.spawn_command_line_sync ("sensors",
```
Comment (ikeydoherty) [inline panel/applets/sensors/SensorsApplet.vala]: There is no `try/catch` here or below

File: panel/applets/sensors/SensorsApplet.vala [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+          if ("unit" == key)
+          {
+                  int unit = m_settings.get_int(key);
+                  switch (unit)
+                  {
+                  case 0 :
+                          m_isCelsius = true;
+                          break;
+
+                  case 1 :
```
Comment (ikeydoherty) [inline panel/applets/sensors/SensorsApplet.vala]:
```
case 1:
default:
    /* Things */
```

File: panel/applets/sensors/SensorsApplet.vala [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+{
+  public Budgie.Applet get_panel_widget(string uuid)
+  {
+          return new SensorsApplet(uuid);
+  }
+}
+
+public class SensorsApplet : Budgie.Applet
+{
+  protected Gtk.EventBox m_widget;
```
Comment (ikeydoherty) [inline panel/applets/sensors/SensorsApplet.vala]: 4 space indentation.

**Comment by ikeydoherty** [comment]
Asides from the inline comments, this should have been at most 2 commits, one to introduce the applet, and a second to then tie it into the main build system (`configure.ac`)

Hope this gives you enough things to work on :)

**Comment by memoryleakx** [comment]
I now have to write this Applet in C ? [https://github.com/solus-project/budgie-desktop/issues/501](url)

**Comment by ikeydoherty** [comment]
No, your PR was created before the change was put in place. Either way it needs to address those comments :)

**Comment by memoryleakx** [comment]
ok! I'm done.. I hope you like it ;)

**Comment by ikeydoherty** [comment]
I think you've missed well over half of my comments as the coding style is still inconsistent and you haven't addressed my locking/concurrency concerns - and I keep seeing merge commits here that I simply *cannot* merge into the repo

**Comment by memoryleakx** [comment]
ok! .. now the code fit with your style guide.

**Pull request closed without merging**